core/include/vecmem/utils/type_traits.hpp

```diff
@@ -113,5 +113,45 @@ auto max(T&& t, Ts&&... ts) {
     return std::max(std::forward<T>(t), max(std::forward<Ts>(ts)...));
 }
 
+/// Type trait that indicates whether a given type is an implicit lifetime
+/// type.
+///
+/// @note The definition of "implicit lifetime type" differs a lot across C++
+/// standards, as does the implementation of this type trait. In C++17 and
+/// earlier, the concept of such types did not exist. In C++20, these types
+/// were defined but no type trait for them was available. In C++23, a type
+/// trait is available.
+///
+/// @warning On pre-C++17 translation units, this type trait is always assumed
+/// to be true.
+///
+/// @{
+#if defined(__cpp_lib_is_implicit_lifetime) && \
+    __cpp_lib_is_implicit_lifetime >= 202302L
+template <class B>
+struct is_implicit_lifetime : std::is_implicit_lifetime<TYPE> {};
```

Comment:
Not sure why you use `B` in all these places, but:

```suggestion
struct is_implicit_lifetime : std::is_implicit_lifetime<B> {};
```

Or better yet, why not

```c++
template <class B>
using is_implicit_lifetime = std::is_implicit_lifetime<B>;
```

?

Reply:
Well spotted, `is_implicit_lifetime` is actually a C++23 feature so not even a C++20 CI build would have caught that. :smile: